File: 0204-count-primes/0204-count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        if n == 1 or n == 0:
            return 0
        def SieveOfEratosthenes(num):
           
            prime = [True for i in range(num)]
        
            prime[0] = prime[1] = False  
        # boolean array


            p = 2
            sqr = math.ceil(math.sqrt(num))
            for p in range(2 , sqr):

                if (prime[p] == True):
                    
                    for i in range(p*p, num, p):
                        prime[i] = False

            return prime
        
        s = SieveOfEratosthenes(n)
        
        count = 0
        for i in s:
            if i:
                count+=1
            
        return count

File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    vector<bool> generated(int n){
        vector<bool> primes(n+1 , true);
        int start = 2;
        primes[0] = false;
        primes[1] = false;
        int sqrtval = ceil(sqrt(n));
        for(long long i=2; i<sqrtval; i++){
            if(primes[i] == true){
                for(long long j = i*i; j<n; j+=i){
                    primes[j] = false;
                }
            }
        }
        
        return primes;
         
        
    }
    int countPrimes(int n) {
        int count = 0;
        vector<bool> primes = generated(n);
        for(int i=0; i<primes.size()-1; i++){
            if(primes[i]) count++;
        }
        return count;
        
    }
};